src/main/java/com/cloudbees/jenkins/support/SupportLogHandler.java

```diff
@@ -234,6 +234,7 @@ private void setFile(File file) throws FileNotFoundException {
                 parentFile.mkdirs();
             }
 
+            StreamUtils.closeQuietly(null); // ensure class is loaded so close() can succeed
```

Comment:
Does nothing but loads this class. Otherwise there can occasionally be a `NoClassDefFoundError` during agent shutdown, at which time it is too late to load new classes from `RemoteClassLoader` because the connection has already been closed.